Content.Shared/Mining/Components/OreVeinComponent.cs

```diff
@@ -23,10 +26,24 @@ public sealed class OreVeinComponent : Component
     [DataField("oreRarityPrototypeId", customTypeSerializer: typeof(PrototypeIdSerializer<WeightedRandomPrototype>))]
     public string? OreRarityPrototypeId;
 
+    /// <summary>
+    /// The weighted random prototype used for determining what ore will be dropped.
+    /// Keyed to a specific tool.
+    /// </summary>
+    [DataField("mappedTools")]
```

Comment:
This needs a serializer but I'm not sure how to do it, this doesn't seem to work
`[DataField("mappedTools", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<WeightedRandomPrototype, TagPrototype>))]`